cli/src/commands/squash.rs

```diff
@@ -155,21 +217,80 @@ pub(crate) fn cmd_squash(
             ));
         }
         sources = vec![source];
-        destination = parents.pop().unwrap();
-    }
+        pre_existing_destination = Some(parents.pop().unwrap());
+    };
 
-    let matcher = workspace_command
+    workspace_command.check_rewritable(sources.iter().chain(&pre_existing_destination).ids())?;
+
+    // prepare the tx description before possibly rebasing the source commits
+    let source_ids: Vec<_> = sources.iter().ids().collect();
+    let tx_description = if let Some(destination) = &pre_existing_destination {
+        format!("squash commits into {}", destination.id().hex())
+    } else {
+        match &source_ids[..] {
+            [] => format!("squash {} commits", source_ids.len()),
+            [id] => format!("squash commit {}", id.hex()),
+            [first, others @ ..] => {
+                format!("squash commit {} and {} more", first.hex(), others.len())
+            }
+        }
+    };
+
+    let mut tx = workspace_command.start_transaction();
+    let destination = if let Some(commit) = pre_existing_destination {
+        commit
+    } else {
+        // create the new destination commit
+        let (parent_ids, child_ids) = compute_commit_location(
+            ui,
+            tx.base_workspace_helper(),
+            args.destination.as_deref(),
+            args.insert_after.as_deref(),
+            args.insert_before.as_deref(),
+            "squash commit",
```

Comment:
nit: "squashed commit" (past tense) or "rebased commit". I'm not sure which one is more correct.